File: Supervised Learning/Regression/Polynomial Regression/coding.py
from sklearn.datasets import load_diabetes
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score, mean_absolute_error, mean_squared_error


diabetes = load_diabetes()
X = diabetes.data
y = diabetes.target

poly_features = PolynomialFeatures(degree=2)

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

X_poly_train = poly_features.fit_transform(X_train)
X_poly_test = poly_features.fit_transform(X_test)

model = LinearRegression()
model.fit(X_poly_train, y_train)
y_predict = model.predict(X_poly_test)


r2 = r2_score(y_test, y_predict)
mae = mean_absolute_error(y_test, y_predict)
mse = mean_squared_error(y_test, y_predict)
rmse = mean_squared_error(y_test, y_predict, squared=False)

print("R-squared score:", r2)
print("Mean Absolute Error:", mae)
print("Mean Squared Error:", mse)
print("Root Mean Squared Error:", rmse)

File: Supervised Learning/Regression/Polynomial Regression/coding.R
library(MASS)

data(Boston)


colnames(Boston)

set.seed(42)

train_indices <- sample(1:nrow(Boston), 0.8 * nrow(Boston))
train_data <- Boston[train_indices,]
test_data <- Boston[-train_indices,]

model <- lm(medv ~ poly(lstat, degree = 2), data = train_data)

prediction <- predict(model, newdata = test_data)

r_squared <- summary(model)$r.squared
mse <- mean((test_data$medv - prediction) ^ 2)
mae <- mean(abs(test_data$medv - prediction))
rmse <- sqrt(mse)

cat("Mean Squared Error:", mse, "\n")
cat("Root Mean Squared Error:", rmse, "\n")
cat("Mean Absolute Error:", mae, "\n")
cat("R-squared:", r_squared, "\n")

